src/Components/Common/ChartSelectionButton/ChartSelectionButton.tsx

```diff
@@ -0,0 +1,42 @@
+import type { FC, ReactElement } from 'react';
+import { NavLink } from 'react-router-dom';
+import { useActions } from '../../../hooks/useActions';
+import { changeType } from '../../../utils/helpers/componentsHelpers';
+import { RoutePath } from '../../../utils/types/enums';
+
+type PropsType = {
+  isLink?: boolean;
+  isDisabled?: boolean;
+  type: string;
+  src: string;
+  text: string;
+  classNameContainer: string;
+  classNameText: string;
+};
+
+export const ChartSelectionButton: FC<PropsType> = ({
+  isLink = false,
+  isDisabled = false,
+  type,
+  src,
+  text,
+  classNameContainer,
+  classNameText,
+}): ReactElement => {
+  const { setMainType, setExamplesType } = useActions();
+
+  const CustomTag = isLink ? NavLink : 'div';
+  const props = {
+    to: isLink ? RoutePath.create : undefined,
+    className: classNameContainer,
+    onClick: isDisabled ? undefined : () => changeType(setMainType, setExamplesType, type),
+  };
+
+  return (
+    // @ts-ignore
```

Comment:
What exactly ts dont like here?

Reply:
`const CustomTag = isLink ? NavLink : 'div';`
`to: isLink && RoutePath.create`  - TS2322: Type { to:  false | RoutePath } is not assignable to type NavLinkProps; Type boolean is not assignable to type To

Reply:
This will give us an empty `to` attribute for the `div` element.